ui/app/routes/vault/cluster/secrets/backend/configuration.js

```diff
@@ -0,0 +1,147 @@
+    switch (type) {
+      case 'aws':
+        return this.fetchAwsConfigs(id);
+      case 'azure':
+        return this.fetchAzureConfig(id);
+      case 'gcp':
+        return this.fetchGcpConfig(id);
+      case 'ssh':
+        return this.fetchSshCaConfig(id);
+    }
+  }
+
+  async fetchAwsConfigs(path) {
+    // AWS has two configuration endpoints root and lease, as well as a separate endpoint for the issuer.
+    const handleError = async (e) => {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 404) {
+        // a 404 error is thrown when the lease config hasn't been set yet.
+        return {};
+      }
+      throw error;
+    };
+
+    const { data: configRoot } = await this.api.secrets
+      .awsReadRootIamCredentialsConfiguration(path)
+      .catch(handleError);
+    const { data: configLease } = await this.api.secrets.awsReadLeaseConfiguration(path).catch(handleError);
+
+    const WIF_FIELDS = ['roleArn', 'identityTokenAudience', 'identityTokenTtl'];
+    const issuer = await this.checkIssuer(configRoot, WIF_FIELDS);
+
+    if (configRoot) {
+      return Object.assign(configRoot, configLease, issuer);
+    }
+  }
+
+  async fetchAzureConfig(path) {
+    try {
+      const { data: azureConfig } = await this.api.secrets.azureReadConfiguration(path);
+      const WIF_FIELDS = ['identityTokenAudience', 'identityTokenTtl'];
+      const issuer = await this.checkIssuer(azureConfig, WIF_FIELDS);
+      // azure config endpoint returns 200 with default values if engine has not been configured yet
+      // all values happen to be falsy so we can just check if any are truthy
+      const isConfigured = Object.values(azureConfig).some((value) => value);
+      if (isConfigured) {
+        return Object.assign({}, azureConfig, issuer);
+      }
+    } catch (e) {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 404) {
+        // a 404 error is thrown when Azure's config hasn't been set yet.
+        return;
+      }
+      throw error;
+    }
+  }
+
+  async fetchGcpConfig(path) {
+    try {
+      const { data: gcpConfig } = await this.api.secrets.googleCloudReadConfiguration(path);
+      const WIF_FIELDS = ['identityTokenAudience', 'identityTokenTtl', 'serviceAccountEmail'];
+      const issuer = await this.checkIssuer(gcpConfig, WIF_FIELDS);
+
+      if (gcpConfig) {
+        return Object.assign(gcpConfig, issuer);
+      }
+    } catch (e) {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 404) {
+        // a 404 error is thrown when GCP's config hasn't been set yet.
+        return;
+      }
+      throw error;
+    }
+  }
+
+  async fetchSshCaConfig(path) {
+    try {
+      const { data } = await this.api.secrets.sshReadCaConfiguration(path);
+      return data;
+    } catch (e) {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 400 && error.errors[0] === `keys haven't been configured yet`) {
```

Comment:
I know this is copied from the original file, but after fixing the bug [here](https://github.com/hashicorp/vault/pull/30838/files#diff-20d17b1773d5de99e4c6d98524e9a3a72b1d6633c4977e891ce0d53f2caf3ea9R155) which was from the go error format changing, it made me realize we should probably make our API error checks a little more resilient in case the go format changes again. 

Just a suggestion, but totally understand if you want to leave as is

```suggestion
      if (error.httpStatus === 400 && errors?.some((str) => str.includes(`keys haven't been configured yet`)) {
```

Reply:
Interesting. Thanks for pointing that out! I suspect this may be an issue in multiple places across the app so it's a good idea to fix the instances as they come up.